src/main.rs

```diff
@@ -355,6 +357,35 @@ async fn main() {
     }
 }
 
+fn respawn_swaps(
+    db: Arc<Database>,
+    bitcoin_wallet: Arc<bitcoin_wallet::Wallet>,
+    ethereum_wallet: Arc<ethereum_wallet::Wallet>,
+    bitcoin_connector: Arc<comit::btsieve::bitcoin::BitcoindConnector>,
+    ethereum_connector: Arc<comit::btsieve::ethereum::Web3Connector>,
+    swap_execution_finished_sender: Sender<FinishedSwap>,
+) -> anyhow::Result<()> {
+    // Assumption: all swaps in the database are active swaps
```

Comment:
Not sure if you already have this in mind, thought I better mention it. 

I think when we re-spawn we will have to do bookkeeping as well, i.e. 

- load swap, evaluate if unfinished
- if unfinished reserve funds in maker
- re-spawn execution with `Sender<FinishedSwap>`
- (free funds logic will be called eventually)

Otherwise we might run into the problem, that we try to free more than is reserved.

Reply:
Added as a TODO.